### PR TITLE
[CODE-3185] Fix the top(N) for the dice roller

### PR DESCRIPTION
**Pull request by kheinzelmann**
When top(N) is specified in the roll(String) method take the top(N) and not the bottom(N) dice.

### CODE REVIEW

**Comment by grimreaper** [comment]
Can we add a test case for this too ?

File: code/src/java/pcgen/core/RollingMethods.java [hunk 1]
```diff
@@ -216,7 +216,7 @@ private static int roll(
         return IntStream.generate(() -> roll(sides - reroll) + reroll)
             .limit(times)
             .sorted()
-            .limit(numToKeep)
+            .skip(times-numToKeep)
```
Comment (grimreaper) [inline code/src/java/pcgen/core/RollingMethods.java]: Another option is to keep this line and replaced .sorted() with `.sorted(Comparator.reverseOrder())`

Comment (kheinzelmann) [inline code/src/java/pcgen/core/RollingMethods.java]: I tried this it does not work with IntStream objects, at least not for me.

**Comment by kheinzelmann** [comment]
I would be willing to attempt adding a test. Do you wont me to add it as another commit to this pull request or open a new pull request?

**Comment by grimreaper** [comment]
> I would be willing to attempt adding a test. Do you wont me to add it as another commit to this pull request or open a new pull request?

I'd prefer another commit on this PR so its more obvious what the test is testing, but that won't block the merge for me.

File: code/src/java/pcgen/core/RollingMethods.java (hunk 1 above)
Comment (grimreaper) [inline code/src/java/pcgen/core/RollingMethods.java]: not a big deal either way. They are similarly readable to me.

**Comment by kheinzelmann** [comment]
Thinking about this I'm not entirely sure how to add a test fixture for this. I would either need to inject the dice rolls as fixed numbers or be able to retrieve the list of generated random numbers. I could possibly cheat if I could set the random number generator seed but that is not currently exposed.

**Comment by grimreaper** [comment]
It would be a good idea to expose the random seed as a Dependency Injection in the ctor. Another option is to look at the existing `DiceTest.randomDiceAssertions` to see how else it mght be done.

**Comment by kheinzelmann** [comment]
DiceTest appears to only check if the dice rolls in the correct range.
RandomUtils is a static class but I should be able to inject via a setter. Let me see what I can do without making a mess of this.

**Pull request merged**